### ISSUE TITLE
Hidden slides flicker when sliding and break through the overflow - Internet Explorer

### ISSUE THREAD

**Issue by DustinJSilk** [kenwheeler/slick#1158]
You can see it here in internet explorer only 
https://www.sanlamreality.co.za/benefit/dining/?mlp-province=Western+Cape&mlp-area=20c58b5f-4f71-49bc-9b5d-cb9f99553992

When sliding the slider, the images outside flicker and show for a few split seconds. I've tried everything. Im sure its a browser rendering issue, but is there a way to fix it?

**Comment by andyrh3** [comment]
Yes I have the same issue where I have the images within a div for cropping purposes. In IE the images flicker when animated! This need fixing asap please. It's a great slider but unfortunately it won't be 'the last slider ever needed' if common bugs like these are not fixed.

**Comment by saress** [comment]
I also have this issue in IE (flickering and breaking through overflow), but it only happens when you use the fade:true mode. I also have tried to use the IE Emulation and it seems to work fine in IE9 and goes wrong in IE10 and Edge.

**Comment by saress** [comment]
As a temporary workaround, this fixes the issue for me:

`
.slick-slider .slick-list, 
.slick-track, 
.slick-slide, 
.slick-slide img { 
    -webkit-transform: translate3d(0, 0, 0); 
    -moz-transform: translate3d(0, 0, 0); 
-ms-transform: none; 
-o-transform: translate3d(0, 0, 0); 
transform: none; 
}
`

Best regards
saress

**Comment by d4v3c** [comment]
I had similar issues, but not limited to images. I think too it's a rendering bug in ie. 

My **solution** was to change the transform method from **translate3d** to **translate**.
(at 3 places in slick.js and once in slick.css. Maybe I will send a PR sometime.)

I didn't have time to read through the code, but I think translate3d is only used to take advantage of hardware acceleration when available.

I too think this is a serious issue. If there is no workaround, then there should be an option to use  translate instead of translate3d.

**Comment by ahmadalfy** [comment]
@d4v3c interesting, can you create a jsfiddle when you have time? I am sure that we are using translate3d for hardware acceleration. Could switch to `will-change` property if that fix it.

**Comment by xkgoldxsite** [comment]
Hi. in google chrome the same problem. After several cycles  image in first  slide is  flickering

**Comment by ninjaonsafari** [comment]
@xkgoldxsite I have the exactly same issue. Have you found a solution?

**Comment by taylorKonigsmark** [comment]
I am seeing the same issue here:

https://cotacchevrolet.com.br/

I can help investigate if you want to get with me @simeydotme

**Comment by ekimbo** [comment]
My images were flicking especially the one in the first slide. I had "cssEase: 'ease-out"  added to the slider and when I commented it out, the flickering stopped. Hope this helps someone out there.

**Comment by DemianD** [comment]
I have the same issue

**Comment by cozzbie** [comment]
Same issue. useCSS: false makes for less smooth transitions. Any other "non-hack" solutions?

**Comment by ddogun** [comment]
.slick-initialized .slick-slide
{
    display: block;
	overflow:hidden;
}
exchange in css !!
I have the same issue ~!

**Comment by bmac1993** [comment]
One of my slides had a an overflow:scroll-y which was causing this issue. Removed it and everything worked fine.

**Comment by RadDog25** [comment]
I had this issue specifically for chrome for a carousel of youtube iframes. Applying these lines of code directly on the iframe worked for me:

transform: translate3d(0, 0, 0);
backface-visibility: hidden;

**Comment by Rhemm** [comment]
I've same issue. And tried a bunch of solutions. And only one which helped me was adding useTransform: false to settings when initialized slider

**Comment by LennonR** [comment]
I had to do useTransform: false in the init options and also 

`.slick-slider .slick-track, .slick-slider .slick-list {
        transform: none;
    }`

**Comment by vishvendrasingh** [comment]
None of above worked in my case. But I checked that the previous of first and next of last element has their id missing so got this working for 3 slides this way - 

	```
        $("[data-slick-index=-1]").find('.success_stories_img').attr('id', 'success_stories_img_3'); //when first time loads for previous slide
	$('.main-class').on('beforeChange', function(event, slick, currentSlide, nextSlide){
		if(nextSlide==0){
			$("[data-slick-index=-1]").find('.class_img').attr('id', 'class_img_3');
		}
		if(nextSlide==2){
			$("[data-slick-index=3]").find('.class_img').attr('id', 'class_img');
		}
	}); 
```

**Comment by PriyaNayak89** [comment]
Kindly stop me from this thread

Thanks,

**Comment by brad1001** [comment]
@LennonR Solution worked for me, but I had a problem with a menu dropdown sitting behind the slideshow after adding the css so I needed to add a z-index like this:

.slick-slider .slick-track, .slick-slider .slick-list { transform: none; z-index: 1; }

Thanks

**Comment by fr0nt3ndd3v** [comment]
If you are using fade and trying to replicate an animated gif with a slider, set .slick-slide to  { transform: none !important; opacity: 1!important; } as the initialized .slick-slide is z-index: 999 and the others are 998. By removing transform and opacity, you can create a seamless animation without flicker. I have sliders within sliders instantiated by iterating through a .each(function) and useCSS didn't seem to fire in that instance.

(function($){
		$(document).ready(function(){

			var carouselOptions = $('._items');
			var mosaicCarousel = '#_HPMosaic_XXXXXX ._carousel ._items';
			var mosaicSlides = (!!$('body').is('.mobile') && 1.5) || 5;
			var innerCarousel = '#_HPMosaic_XXXXXX ._carousel ._items ._item ._items';
			var innerSlides = 1;

			$(mosaicCarousel, innerCarousel).slick({
				lazyLoad:'ondemand',
				slidesToScroll: 1,
				infinite: false,
				arrows: false,
				dots: false
			});	
			
			carouselOptions.each(function() {
				if ($(this).is('._items._mc')) {
					$(this).slick({
						slidesToShow: mosaicSlides,
						autoplay: false,
						autoplaySpeed: 0,
						swipeToSlide: true,
					});
				} 
				else if ($(this).is('._items._ic')){
					$(this).slick({
						slidesToShow: innerSlides,
						autoplay: true,
						autoplaySpeed: 1000,
						fade: true,
						useCSS: false,
						swipeToSlide: false,
					});
				}
				else {
					$(this).slick();
				}
			});

		});

		$('#_HPMosaic_XXXXXX').addClass('_loaded');

	})(jQuery);
